core/src/com/unciv/ui/mapeditor/TileEditorOptionsTable.kt

```diff
@@ -149,48 +153,92 @@ class TileEditorOptionsTable(val mapEditorScreen: MapEditorScreen): Table(Camera
+                                        tile.improvement = null
+                                    tile.setTerrainTransients()
+                                    tileGroup.update()
+                                }
+                            }
+
+                            val nationIcon = getHex(Color.WHITE, ImageGetter.getNationIndicator(nation, 40f))
+                            setCurrentHex(nationIcon,"Player $playerIndex starting location")
+                        }
+                        nationTable.add(nationImage).row()
                     }
                 }
 
-                val nationIcon = getHex(Color.WHITE, ImageGetter.getNationIndicator(nation, 40f))
-                setCurrentHex(nationIcon, "[${nation.name}] starting location")
+        } else {
+            /** old way improvements for all civs
+             * */
+            for(nation in ruleset.nations.values){
+                val nationImage = getHex(Color.WHITE, ImageGetter.getNationIndicator(nation, 40f))
+                nationImage.onClick {
+                    val improvementName = "StartingLocation "+nation.name
+
+                    tileAction = {
+                        it.improvement = improvementName
+                        for (tileGroup in mapEditorScreen.mapHolder.tileGroups.values) {
+                            val tile = tileGroup.tileInfo
+                            if (tile.improvement == improvementName && tile != it)
+                                tile.improvement = null
+                            tile.setTerrainTransients()
+                            tileGroup.update()
+                        }
+                    }
+
+                    val nationIcon = getHex(Color.WHITE, ImageGetter.getNationIndicator(nation, 40f))
+                    setCurrentHex(nationIcon, "[${nation.name}] starting location")
+                }
+                nationTable.add(nationImage).row()
             }
-            nationsTable.add(nationImage).row()
         }
 
-        editorPickTable.add(AutoScrollPane(nationsTable)).height(scrollPanelHeight)
+        editorPickTable.add(AutoScrollPane(nationTable).apply { setScrollingDisabled(true,false) }).height(scrollPanelHeight)
     }
 
     fun setUnits(){
         editorPickTable.clear()
-        var currentNation = ruleset.nations.values.first()
+
+        val nationsTable = Table()
+
+        // default player - first non-random player or barbarians
+        var currentPlayer = ""
+        var currentNation: Nation? = ruleset.nations.values.firstOrNull{ it.isBarbarian() }
+
         var currentUnit = ruleset.units.values.first()
+
         fun setUnitTileAction(){
-            val unitImage = ImageGetter.getUnitIcon(currentUnit.name, currentNation.getInnerColor())
-                    .surroundWithCircle(40f).apply { color=currentNation.getOuterColor() }
-            setCurrentHex(unitImage, currentUnit.name.tr()+" - "+currentNation.name.tr())
+            if (currentNation == null) return
```

Comment:
If you added 'if (currentNation == null) return", then the compiler should recognize that all the below !! on currentNation!! are unneeded :)

Reply:
Don't know why, but my compiler doesn't recognize it :( And doesn't want to compile it with just currentNation.name. Maybe I additionally configure it?

Reply:
Nah, it's not that important